File: backend/app/schemas/agent_node.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AgentNodeResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    task_id: int
    node_key: str
    node_name: str
    node_type: str
    status: str
    input_summary: str | None = None
    output_summary: str | None = None
    started_at: datetime | None = None
    ended_at: datetime | None = None
    duration_ms: int | None = None
    retry_count: int
    error_message: str | None = None


class DagEdge(BaseModel):
    source: str
    target: str


class AgentNodeListResponse(BaseModel):
    nodes: list[AgentNodeResponse]
    edges: list[DagEdge]


class AgentLogResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    task_id: int
    node_id: int | None
    log_type: str
    message: str
    payload: dict | None = None
    created_at: datetime


class AgentLogListResponse(BaseModel):
    logs: list[AgentLogResponse]

File: frontend/src/types/agentNode.ts
export interface AgentNode {
  id: number
  task_id: number
  node_key: string
  node_name: string
  node_type: string
  status: string
  input_summary?: string | null
  output_summary?: string | null
  started_at?: string | null
  ended_at?: string | null
  duration_ms?: number | null
  retry_count: number
  error_message?: string | null
}

export interface DagEdge {
  source: string
  target: string
  type?: string
  label?: string | null
}

export interface AgentLog {
  id: number
  task_id: number
  node_id?: number | null
  log_type: string
  message: string
  payload?: Record<string, unknown> | null
  created_at: string
}
